### PR TITLE
#1147: separate plugins from software

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Addresses/Fixes: #1147 

Implements:

* changed plugin paths from software/x-plugin to plugins/x
* adjusted documentation

### CODE REVIEW

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -165,7 +167,7 @@ New release with various improvements and bugfixes:
 * https://github.com/devonfw/ide/issues/692[#692]: Improve Windows Terminal with gitbash
 * https://github.com/devonfw/ide/issues/786[#786]: Improve Windows Terminal with powershell
 * https://github.com/devonfw/ide/issues/849[#849]: Integration of GCViewer
-* https://github.com/devonfw/ide/issues/867[#867]: ERROR: Error opening the file. There may be a disk or file system error.
+* https://github.com/devonfw/ide/issues/867[#867]: ERROR: Error opening the file, there may be a disk or file system error
```
Comment (hohwille) [inline CHANGELOG.asciidoc:170]: are you using auto-magic AsciiDoc plugins that try to do "auto-correction".
This is the summary of an issue and it comes from an actually error message that is just the way this error message is.
Chaning this title here does not seem to make sense.
Tame your IDE :)

**Review by hohwille — COMMENTED** [review]
@jan-vcapgemini thanks for your PR. Even though still in draft state this looks fine for me. 👍 
We should then also update #1135 accordingly.
For me it makes sense to ship this change with the next release that brings more of such structural changes. Further we should also add this to our migration script (see #1123).
Could you include the migration script addition to this PR (and also have a look at my review comment)? Thanks

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (jan-vcapgemini) [reply CHANGELOG.asciidoc]: I've tried to manually fix the conflict between end of lines with a dot and a required newline.
It's reverted now.

**Comment by jan-vcapgemini** [comment]
> @jan-vcapgemini thanks for your PR. Even though still in draft state this looks fine for me. 👍 We should then also update #1135 accordingly. For me it makes sense to ship this change with the next release that brings more of such structural changes. Further we should also add this to our migration script (see #1123). Could you include the migration script addition to this PR (and also have a look at my review comment)? Thanks

I've adjusted the CHANGELOG.asciidoc and migration script now.
I was waiting for the changes of: #1135, but we can also update the paths there too.

File: scripts/src/main/resources/scripts/migrations/2023.06.001 [hunk 2]
```diff
@@ -47,3 +47,22 @@ find "${DEVON_IDE_HOME}"/workspaces/ -mindepth 2 -maxdepth 2 -name devon.propert
...
+plugins_path="${DEVON_IDE_HOME}/plugins"
+
+if [ ! -d "${plugins_path}" ]
+then
+	mkdir "${plugins_path}"
+else
+	doWarning "Plugins directory already exists!"
+fi
+
+find "${DEVON_IDE_HOME}"/software -maxdepth 1 -type d -name "*-plugins" -print | while read found_folder
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/migrations/2023.06.001:60]: Great that you added the migration. 👍 
As we only have 2 specific candidates here (`vscode-extensions` and `eclipse-plugins`) it does not make much sense trying to solve this in a generic way esp. as `vscode-extensions` does not match `*-plugins`.
I would simply hardcode these two folders here. In the end all we need is just
```
mv "${DEVON_IDE_HOME}/software/vscode-extensions" "${DEVON_IDE_HOME}/plugins/vscode"
mv "${DEVON_IDE_HOME}/software/intellij-plugins" "${DEVON_IDE_HOME}/plugins/intellij"
```

Android-studio has not yet been released and eclipse is not yet separated so KISS is key.

Comment (hohwille) [reply scripts/src/main/resources/scripts/migrations/2023.06.001:60]: Feel free to check the directories for existence to suppress `mv: cannot stat 'software/vscode-extensions': No such file or directory` error.

Comment (jan-vcapgemini) [reply scripts/src/main/resources/scripts/migrations/2023.06.001]: I've applied KISS now and added checks for existing directories.

**Review by hohwille — APPROVED** [review]
@jan-vcapgemini excellent. Thanks for the update 👍 
Ready for merge.

**Pull request merged**